File: pset2/caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])
{
    char ct;
    if (argc == 0 || argc < 2)
    {
        printf("error");
        return 1;
    }
    else
    {
       int k = atoi(argv[1]);
       printf("plaintext: ");
       string pt = get_string();
       printf("ciphertext: ");
       for ( int i = 0, n = strlen(pt); i < n; i++ )
       if (isalpha(pt[i]))
       {
           if ( isupper(pt[i]))
           {
            ct = (pt[i] -'A' + k) % 26 + 'A';
           }
           else if ( islower(pt[i]))
           {
            ct = (pt[i] - 'a' + k) % 26 + 'a';
           }
        printf("%c", ct);
       }
       else
       {
           ct = pt[i];
       }
       printf(" ");
    }
        printf("\n");

}

File: pset6/caesar/caesar.py
from cs50 import get_int
from cs50 import get_string
import cs50
import sys

# caesars cipher

key = int(sys.argv[1])

plainText = get_string("plaintext: ")  # prompts user for string to encrypt
print("ciphertext: ", end="")

for c in range(len(plainText)):  # iterate through each character
    character = plainText[c]
    if character.isalpha():  # check if character is alphabetic
        if character.isupper():  # preserve case of character
            cipherText = (ord(character) - ord('A') + key) % 26 + ord('A')  # to ascii
            cipherText = chr(cipherText)  # to alphabetic
            print(cipherText, end="")
        elif character.islower():
            cipherText = (ord(character) - ord('a') + key) % 26 + ord('a')  # to ascii
            cipherText = chr(cipherText)  # to alphabetic
            print(cipherText, end="")
    elif not plainText[c].isalpha():  # includes special characters
        cipherText = character
        print(cipherText, end="")
print()
